### DISCUSSION TITLE
Tomo - creating sinograms to a 3d volume with anisotropic voxels

### DISCUSSION THREAD

**Discussion by orshaf** [astra-toolbox/astra-toolbox#402] [Q&A] [answered]
Hello

I am creating sinograms using create_sino3d_gpu.
I have 3d volume as an input with anisotropic voxels. 
I have the detector size in mm of pixels and distance from the patient, etc. 
Now, I don't know how to tell astra_toolbox the size of the voxels of the volume. I guess it assumes that is isotropic 1 mm voxels, because if I insert for example slice thickness of 2 mm coronal view, then the volume looks like someone squeezed it into half. 
I tried to change vol_geom, but then the create_sino3d_gpu() raises an error because the dimensions of vol_geom is different of the volume. 

Please help me what am i missing :)

**Comment by askorikov** [comment]
Can you provide a minimal example of the problem? Not necessarily with your data, just illustrate the dimensions you're using.

**Comment by orshaf** [comment]
astra.creators.create_sino3d_gpu(data, proj_geom, vol_geom, returnData=True, gpuIndex=None)
data = nd.array (512, 365, 512)
vol_geom = {'option': {'WindowMinX': -256, 'WindowMaxX': 256, 'WindowMinY': -200, 'WindowMaxY': 200, 'WindowMinZ': -256, 'WindowMaxZ': 256}, 'GridRowCount': 400, 'GridColCount': 512, 'GridSliceCount': 512}
proj_geom = self.proj_geom
{'type': 'cone_vec', 'DetectorRowCount': 3000, 'DetectorColCount': 3000, 'Vectors': array([[ 0.00000000e+00, -2.07911691e+02,  7.22147601e+02,
         0.00000000e+00,  0.00000000e+00, -3.75000000e+02,
         1.40000000e-01,  0.00000000e+00,  0.00000000e+00,
         0.00000000e+00,  1.36940664e-01,  2.91076367e-02],
       [ 0.00000000e+00, -1.93763504e+02,  7.25048268e+0 ... etc. pixel size of detector is 0.14 mm

**Reply by askorikov** [reply]
In your case `vol_geom['GridRowCount']` should be the same as `data.shape[1]`.

**Comment by orshaf** [comment]
I understand that. but where do i insert the voxel size?

**Reply by askorikov** [reply, marked as answer]
It's determined by the `WindowMin...`, `WindowMax...` for every dimension, which you can set like explained in the [original post  ](https://github.com/astra-toolbox/astra-toolbox/discussions/401#discussioncomment-6897223). Voxel size is then `(WindowMax - WindowMin) / n_voxels`. Basically, you need to adjust `min_x` and `max_x` by a factor, if I understood your case correctly.

**Comment by orshaf** [comment]
It worked!!
Thank you very much
